File: ComponentStudy/opengl/src/main/java/com/example/opengl/MainActivity.java
package com.example.opengl;

import android.os.Bundle;

import com.example.opengl.databinding.ActivityMainBinding;

public class MainActivity extends PermissionActivity {

    private ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.customSurfaceView.setActivity(MainActivity.this);
    }
}

File: AndroidSdkStudy/app/src/main/java/com/vtech/mobile/androidsdkstudy/MainActivity.kt
package com.vtech.mobile.androidsdkstudy

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.vtech.mobile.androidsdkstudy.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.sampleText.text = stringFromJNI()
    }

    /**
     * A native method that is implemented by the 'androidsdkstudy' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    companion object {
        // Used to load the 'androidsdkstudy' library on application startup.
        init {
            System.loadLibrary("androidsdkstudy")
        }
    }
}